src/include/wiredtiger.in

```diff
@@ -1937,9 +1944,36 @@ struct __wt_session {
 	 */
 	int __F(timestamp_transaction)(WT_SESSION *session, const char *config);
 
+	/*!
+	 * Set a timestamp on a transaction.
+	 *
+	 * Takes the timestamp as a numeric value in the form of wt_timestamp_t.
+	 *
+	 * @snippet ex_all.c transaction timestamp numeric
+	 *
+	 * @requires_transaction
+	 *
+	 * @param session the session handle
+	 * @param timestamp a wt_timestamp_t type, timestamp value being set.
+	 * @configstart{WT_SESSION.timestamp_transaction_numeric, see
+	 * dist/api_data.py}
+	 * @config{set, specify which timestamp to set. Refer to
+	 * WT_SESSION.timestamp_transaction for details.  See @ref
+	 * transaction_timestamps., a string\, chosen from
+	 * the following options: \c "commit_timestamp"\,
+	 * \c "durable_timestamp"\, \c "prepare_timestamp"\,
+	 * \c "read_timestamp".}
+	 * @configend
+	 * @errors
+	 */
+	int __F(timestamp_transaction_numeric)(
```

Comment:
Apologies if I've remarked on the wrong method/line:

Suppose I have a transaction in a prepared state. How would I write the analogous code for:
```
session->timestamp_transaction(session, "durable_timestamp=20,commit_timestamp=10");
```

Reply:
That would need to become two calls to the new API:
```
session->timestamp_transaction_numeric(session, 20, "set=durable_timestamp");
session->timestamp_transaction_numeric(session, 10, "set=commit_timestamp");
```

We intend to keep both versions of the API - I suspect there are cases where making a single call to WiredTiger is preferrable to passing in numeric timestamp arguments.